src/main/java/bio/terra/workspace/service/job/JobService.java

```diff
@@ -260,23 +249,17 @@ public FlightState call() throws Exception {
    * encapsulates all of the Stairway interaction.
    */
   public void initialize() {
-    try {
-      stairway.initialize(
-          stairwayDatabaseConfiguration.getDataSource(),
-          stairwayDatabaseConfiguration.isForceClean(),
-          stairwayDatabaseConfiguration.isMigrateUpgrade());
-      stairway.recoverAndStart(null);
-
-    } catch (StairwayException | InterruptedException stairwayEx) {
-      throw new InternalStairwayException("Stairway initialization failed", stairwayEx);
-    }
+    stairwayComponent.initialize(
+        stairwayDatabaseConfiguration.getDataSource(),
+        flightBeanBag,
+        ImmutableList.of(mdcHook, new TracingHook()));
   }
 
   @Traced
   public void releaseJob(String jobId, AuthenticatedUserRequest userReq) {
     try {
       verifyUserAccess(jobId, userReq); // jobId=flightId
-      stairway.deleteFlight(jobId, false);
+      stairwayComponent.get().deleteFlight(jobId, false);
```

Comment:
should the stairway component just expose these functions instead of exposing the stairway object with get()?

Reply:
I don't suggest that. Feels we are putting to much into stairwayComponent, and make it become part of 'Stairway library' which is unnecessary. And if Stairway method changes, we have to upgrade stairwayComponent correspondingly. 
Using stairwayComponent just for initialization and health status should be good enough.

Reply:
👍🏻

Reply:
I agree with @yonghaoy.
We could capture the Stairway object into a `private final Stairway stairway` from the stairwayComponent in the JobService constructor and keep using the `stairway` variable, right?

Reply:
You can't initialize a final field except in the constructor, or at least it's a warning.

I'm concerned we don't have a clean abstraction. Either `StairwayComponent` owns `Stairway`, or it doesn't. I don't like `get()`, but I like saving the result of a `get()` even less. It doesn't _belong_ to `JobService`.

Should that method be removed from `StairwayComponent`, and we just return a `Stairway` from the initialize method so it's like a factory? That approach wouldn't make sense with the status, unless we assume `Stairway` really is a singleton.

Reply:
Sorry if that sounded aggresive.